src/canvas_classes/square.class.ts

```diff
@@ -8,26 +8,26 @@ import { Tool } from './tool.class'
 
 export class Square extends Tool {
   private mouseDown = false
+  private saved = ''
   private x1 = 0
   private y1 = 0
-  private saved = ''
 
-  constructor(canvas: React.MutableRefObject<HTMLCanvasElement>, socket: SocketApp, id: string) {
+  constructor(canvas: HTMLCanvasElement, socket: SocketApp, id: string) {
     super(canvas, socket, id)
     this.listen()
   }
 
   private listen() {
-    this.canvas.current.onmousedown = this.onMouseDown.bind(this)
-    this.canvas.current.onmousemove = this.onMouseMove.bind(this)
-    this.canvas.current.onmouseup = this.onMouseUp.bind(this)
+    this.canvas.onmousedown = this.onMouseDown.bind(this)
+    this.canvas.onmousemove = this.onMouseMove.bind(this)
+    this.canvas.onmouseup = this.onMouseUp.bind(this)
   }
 
   private onMouseDown(e: MouseEvent) {
     this.mouseDown = true
     this.x1 = e.offsetX
     this.y1 = e.offsetY
-    this.saved = this.canvas.current.toDataURL()
+    this.saved = this.canvas.toDataURL()
   }
```

Comment:
How many times you have this strings in your project?

Reply:
classes also can be extendable

Reply:
this.saved should be not to all classes